src/libxtp/threecenter_gwbse.cc

```diff
@@ -150,15 +176,55 @@ void TCMatrix_gwbse::FillBlock(std::vector<Eigen::MatrixXd>& block,
       }
     }  // gamma-loop
   }    // alpha-loop
-  for (int k = 0; k < auxshell.getNumFunc(); ++k) {
+
+#if defined(USE_GPU)
+  // create tensor
+  std::vector<Eigen::MatrixXd> tensor;
+  for (const Eigen::MatrixXd& mtx : symmstorage) {
+    tensor.push_back(mtx.selfadjointView<Eigen::Lower>());
```

Comment:
how does the conversion from selfadjoint view to full Matrix work?

Reply:
I have replaced the `push_back` method with `emplace_back`. The reason is that `emplace_back`  uses *perfect forwarding* that takes `constructor arguments for objects to be inserted` while `push_back`  takes ` objects to be inserted`.  This difference allows `emplace_back` to avoid the  creation and destruction of temporary objects